File: TypeScript/1456. Maximum Number of Vowels in a Substring of Given Length.ts
// noinspection JSUnusedGlobalSymbols
const maxVowels = (s: string, k: number): number => {
  let max = 0

  for (let i = 0; i < k; i++) {
    if (vowelCheck(s[i])) {
      max++
    }
  }

  let current = max
  for (let i = k; i < s.length; i++) {
    if (vowelCheck(s[i - k])) {
      current--
    }

    if (vowelCheck(s[i])) {
      current++
    }

    if (current > max) {
      max = current
    }
  }

  return max
}

const vowelCheck = (char: string): boolean => char == 'a' || char == 'e' || char == 'i' || char == 'o' || char == 'u'

File: C-Sharp/1456. Maximum Number of Vowels in a Substring of Given Length.cs
using System.Diagnostics.CodeAnalysis;

namespace LeetCode.C_Sharp;

[SuppressMessage("ReSharper", "UnusedType.Global")]
[SuppressMessage("ReSharper", "UnusedMember.Global")]
public partial class Solution
{
    public static int MaxVowels(string s, int k)
    {
        int maxVowel = 0;

        for (int i = 0; i < k; i++)
        {
            if (VowelCheck(s[i]))
            {
                maxVowel++;
            }
        }

        int current = maxVowel;
        for (int i = k; i < s.Length; i++)
        {
            if (VowelCheck(s[i - k]))
            {
                current--;
            }

            if (VowelCheck(s[i]))
            {
                current++;
            }

            if (current > maxVowel)
            {
                maxVowel = current;
            }
        }

        return maxVowel;
    }

    private static bool VowelCheck(char c)
    {
        return c is 'a' or 'e' or 'i' or 'o' or 'u';
    }
}
